frontend/ui/src/app/components/quota-limit.component.ts

```diff
@@ -29,11 +30,12 @@ export class QuotaLimitComponent {
   protected readonly percentage = computed(() => {
     const u = this.usage();
     const l = this.limit();
-    if (l === undefined) {
+    if (l === undefined || l < 0) {
```

Comment:
I think either we should check against the "unlimited" constant everywhere or nowhere. Right now it is inconsistent